be/src/io/cache_input_stream.cpp

```diff
@@ -208,6 +208,7 @@ Status CacheInputStream::read_at_fully(int64_t offset, void* out, int64_t count)
 }
 
 StatusOr<int64_t> CacheInputStream::read(void* data, int64_t count) {
+    count = std::min(_size - _offset, count);
```

Comment:
So it means the caller in csv scanner doesn't ensure the read range is not exceeded ?

Reply:
Sure. because csv scanner decides if it reaches EOF by looking at return value of `read` ( if ret = 0).